### DISCUSSION TITLE
Permissions dialog switches out app after approving

### DISCUSSION THREAD

**Discussion by Schwirzke** [yasirkula/UnitySimpleFileBrowser#98] [Q&A]
Android 12+, On fresh install when the open file dialog is shown and I click on browse the permissions dialog comes up, I approve the directory but then I'm no longer in my app, all I see is a dark screen and the Occulus app menu on the bottom, if I hit the home button on my controller or click on my app icon in the oculus bar,  it takes me back to my app and everything works fine, is there a way to avoid this "black screen" limbo after granting permissons?

**Comment by yasirkula** [comment]
Can you try applying the Oculus Quest related change mentioned in FAQ: https://github.com/yasirkula/UnitySimpleFileBrowser?tab=readme-ov-file#faq

**Reply by Schwirzke** [reply]
I don't see anything in the FAQ relating to this issue... everything works besides this focus issue, it's basically the issue this guy is having https://communityforums.atmeta.com/t5/Unity-VR-Development/Oculus-Quest-Need-to-Dismiss-Oculus-Menu-Bar-After-Permissions/td-p/1047113

**Reply by yasirkula** [reply]
Have you actually tried the 3rd FAQ entry that specifically mentions Quest?

**Reply by Schwirzke** [reply]
This is what I see as the 3rd entry in the FAQ, is this the one I should be looking at?

**File browser doesn't show any files on Oculus Quest**.   

Is this the one you are referring do? Note all the files show after granting permissions it works great and I love it!  I copy the file over to my app directoy using the helper routines provided then I can unzip them and everything works great. Well done!

However  that's not the problem I'm having. I see nothing in the FAQ about the oculus home menu bar being active instead of the app after granting permissions the first time.

**Reply by yasirkula** [reply]
Yes that's the one. I wish, considering my consistent persuasion attempts, you had given it a try. That fix removes the "Browse..." button and displays the files without requiring the native dialog that "Browse..." used to present.

**Reply by Schwirzke** [reply]
I gave it a try implementing fix #87 and #89 , it did remove the browse button and the native browse/permisssion dialog, but I get a different system dialog to enable file access to everything, on giving approval it goes away, but my app still does not get focus back. On pressing the re center button my app becomes active and I can browse and load files as before. The only difference is I"m not seeing a black screen behind the system menu bar, but my app instead, but it does not respond to anything until I hit the re center button as before. 

I will probably go back to the browse button permissions way, as the permissions dialog is not as scary sounding as the new one which implies I'm getting access to every file on the device, which I suppose is what's happening when requesting ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION as suggested in fix #89

Thanks for the quick response, you make some high quality tools, thank you.

**Reply by yasirkula** [reply]
Hmm, how about implementing only #87? That may not grant access to all files under Downloads (as suggested by #89) but maybe those files can't be browsed with "Browse..." as well so we may not be losing anything when compared to that.

I wish I knew a proper solution for the black screen limbo though. My other plugin NativeFilePicker also suffers from it.
